Expand the section on type signatures in the README (#349)
fluture-js/Fluture: README.md

### Type signatures

The various function signatures are provided in a small language referred to as
Hindley-Milner notation.

In summary, the syntax is as follows: `InputType -> OutputType`. Now,
because functions in Fluture are [curried][Guide:currying], the "output" of a
function is often *another function*. In Hindley-Milner that's simply written
as `InputputType -> InputToSecondFunction -> OutputType` and so forth.

> By convention, types starting with a capital letter are

Reviewer: This might be clearer to non-native English speakers:

```suggestion
By convention, types starting with an upper-case letter are
```

[concrete types](#types). When they start with a lower-case letter they're
*type variables*. You can think of these type variables as generic types.
So `a -> b` denotes a function from generic type `a` to generic type `b`.

Finally, through so-called [*constraints*](#type-classes), type variables can
be forced to conform to an "interface" (or *Type Class* in functional jargon).
For example, `MyInterface a => a -> b`, denotes a function from generic type

> `a` to generic type `b`, *where `a` must have `MyInterface` implemented*.

Reviewer: ```suggestion
`a` to generic type `b`, *where `a` must implement `MyInterface`*.
```
